index.ts

```diff
@@ -72,9 +73,16 @@ async function init() {
 	}
 
 	input.addEventListener('input', () => {
-		localStorage.setItem('token', input.value);
+		if (input.validity.valid) {
+			localStorage.setItem('token', input.value);
+		}
```

Comment:
I'm not a fan of fields that are not saved when invalid. It's best to still save it and display that it's invalid (as it is)